api/osb/catalog_fetcher.go

```diff
@@ -0,0 +1,34 @@
+package osb
+
+import (
+	"context"
+
+	"github.com/Peripli/service-manager/pkg/types"
+	"github.com/Peripli/service-manager/storage"
+)
+
+// StorageCatalogFetcher fetches the broker's catalog from SM DB
+type StorageCatalogFetcher struct {
+	CatalogStorage storage.ServiceOffering
+}
+
+// FetchCatalog implements osb.CatalogFetcher and fetches the catalog for the broker with the specified broker id from SM DB
+func (scf *StorageCatalogFetcher) FetchCatalog(ctx context.Context, brokerID string) (*types.ServiceOfferings, error) {
+	catalog, err := scf.CatalogStorage.ListWithServicePlansByBrokerID(ctx, brokerID)
+	if err != nil {
+		return nil, err
+	}
+
+	// SM generates its own ids for the services and plans - currently for the platform we want to provide the original catalog id
```

Comment:
Idea for another PR: Is it possible to just change the names of our ids and set the catalog ids as just ID?